ytmp3/downloader.py

```diff
@@ -36,29 +33,34 @@ def __init__(self, bot, logging, options):
     async def ping(self, ctx):
         if not ctx.guild:
             return
-
         self.__log(f"User {ctx.author.id} pinged inside guild {ctx.guild.id}")
 
         await ctx.reply("Pong")
 
     @commands.command()
     async def download(self, ctx, url):
-        """
-        Downloads video from provided URL,
-        reply directly to discord user with URL of converted .mp3 file
-        """
         # Don't serve via direct message
         if not ctx.guild:
             return
 
         self.__log(f"User {ctx.author.id} attempted to download {url} inside guild ID {ctx.guild.id}")
 
         # Check URL for nasties like &list=
-        containsList = not not re.search("list=", url)
+        containsList = bool('list=' in url)
 
         if containsList:
             self.__warning("Detected 'list=' substring inside the URL.")
-            return await ctx.reply("Cannot accept URL containing 'list='")
+            listurl = url.split('list=')[0]
```

Comment:
This makes a (potentially) dangerous assumption that the user will always provide sane urls. The query parameters in the URL don't need to be in any particular order.

Think what might happen when you give https://www.youtube.com/watch?list=PLqXoOPKEpA0_BKqpeDxcHOVWwcmxZRp_-&v=KNEiV5hNUAo

It would probably cause an exception. You have to be really careful about the input you process from strangers online.

I think it would be better to use string.replace or something similar to remove  `list=` and it's value completely rather than using as an anchor split the string.

So the example above,
https://www.youtube.com/watch?list=PLqXoOPKEpA0_BKqpeDxcHOVWwcmxZRp_-&v=KNEiV5hNUAo

would become
https://www.youtube.com/watch?v=KNEiV5hNUAo

In this case you are actually replacing `list=....&` so you're probably better off using regex.